lib/elixir/lib/uri.ex

```diff
@@ -204,9 +208,9 @@ defmodule URI do
       true
 
   """
-  @spec char_reserved?(term) :: boolean
-  def char_reserved?(c) do
-    c in ':/?#[]@!$&\'()*+,;='
+  @spec char_reserved?(pos_integer) :: boolean
+  def char_reserved?(char) when is_integer(char) and char > 0 do
+    char in ':/?#[]@!$&\'()*+,;='
   end
```

Comment:
Once we have 

```elixir
def char_reserved?(char) when char in 0..0x10ffff do
  char in ':/?#[]@!$&\'()*+,;='
end
```

I think we'll be ok. Yes, the first check could be avoided with your solution, but it's still a very very simple check (it will expand to just a `<` and `>`). I'd trade that for a cleaner-looking function, but we can wait on someone else's opinion on this as well.